### PR TITLE
[CosmosDB] Updated to version 2023-03-15-preview

### PR DESCRIPTION
**Pull request by SrinikhilReddy**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
CosmosDB

**Comment by evelyn-ys** [comment]
Please make sure all CI checks can pass

**Comment by SrinikhilReddy** [comment]
> Please make sure all CI checks can pass

All gates passing please rereview

File: src/cosmosdb-preview/azext_cosmosdb_preview/_help.py [hunk 1]
```diff
@@ -796,6 +796,24 @@
                az cosmosdb mongodb collection merge -g my-resource-group -a my-account -d my-db --name my-mongodb-collection
 """
 
+helps['cosmosdb sql database merge'] = """
+    type: command
+    short-summary: "Merges the partitions of a sql database."
+    examples:
+      - name: merge partitions of database my-database
```
Comment (evelyn-ys) [inline src/cosmosdb-preview/azext_cosmosdb_preview/_help.py:803]:
```suggestion
    short-summary: "Merge the partitions of a sql database."
    examples:
      - name: Merge partitions of database my-database
```

**Review by evelyn-ys — COMMENTED** [review]
It shows like you delete all `_param.py`, `commands.py` and `custom.py` and recreate again......

It's really hard to review what you actually changed

**Comment by SrinikhilReddy** [comment]
> It shows like you delete all `_param.py`, `commands.py` and `custom.py` and recreate again......
> 
> It's really hard to review what you actually changed

image

Using this option ignores the white space changes.

File: src/cosmosdb-preview/azext_cosmosdb_preview/tests/latest/test_cosmosdb-merge-scenario.py [hunk 2]
```diff
@@ -1,75 +1,135 @@
...
+
+        # update container
+        self.cmd('az cosmosdb sql database throughput update -g {rg} -a {acc} -n {db_name} --throughput 3000').get_output_in_json()
+
+        # merge
+        # Currently this should throw as the feature is disabled
+        try:
+            self.assertRaises(HttpResponseError, lambda: self.cmd('az cosmosdb sql database merge -g {rg} -a {acc} -n {db_name} '))
+        except Exception as err:
+            self.assertTrue("Merge operation feature is not available" in err)
```
Comment (evelyn-ys) [inline src/cosmosdb-preview/azext_cosmosdb_preview/tests/latest/test_cosmosdb-merge-scenario.py:103]: Please add CLI support after feature ready for use

**Review by evelyn-ys — COMMENTED** [review]
If you want to release a new version, pls update the version in `setup.py` and `histroy.rst`

**Review by evelyn-ys — APPROVED** [review]

**Pull request merged**